### PR TITLE
Prerelease v0.11.0-alpha

### PR DESCRIPTION
**Pull request by damemi**
### Added

- Test build using [Go 1.22]. (#672)
- Base Dockerfile and build caching ([#683](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/683))
- Add `server.address`, `server.port` and `network.protocol.version` to HTTP client spans ([#696](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/696))
- Update http server attributes to latest semantic conventions ([#708](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/708))
### Fixed
- Don't call `manager.Close()` when `Analyze()` fails. ([#638](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/638))
- Close `proc` file when done discovering PID. ([#649](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/649))
- Use `debug` packages to parse Go and modules' versions. ([#653](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/653))
- Clean up warn in otelglobal `SetStatus()` when grabbing the status code. ([#675](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/675))
- Reset `proc` offset after a failed iteration. ([#681](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/681))
- Avoid using runtime.NumCPU to get the number of CPUs on the system before remote mmap ([#680](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/680))
- Cleanup eBPF maps only when we stop using the memory ([#682](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/682))

[Go 1.22]: https://go.dev/doc/go1.22

### CODE REVIEW

**Comment by damemi** [comment]
@MrAlias I had to manually update `version.go`, is `make prerelease` supposed to do that?

**Comment by RonFed** [comment]
Hi @damemi, I have a small bug fix I wish to get merged before this release, I will open a PR in the next hour or so.

**Comment by MrAlias** [comment]
> @MrAlias I had to manually update `version.go`, is `make prerelease` supposed to do that?

No, you did it correctly. That is a manual process given it is not always clear what version you are bumping to for each modset.

File: CHANGELOG.md [hunk 1]
```diff
@@ -287,7 +293,8 @@ OpenTelemetry Go Automatic Instrumentation adheres to [Semantic Versioning](http
 
 This is the first release of OpenTelemetry Go Automatic Instrumentation.
 
-[Unreleased]: https://github.com/open-telemetry/opentelemetry-go-instrumentation/compare/v0.10.1-alpha...HEAD
+[Unreleased]: https://github.com/open-telemetry/opentelemetry-go-instrumentation/compare/v0.11.0-alpha...HEAD
+[v0.10.1-alpha]: https://github.com/open-telemetry/opentelemetry-go-instrumentation/releases/tag/v0.11.0-alpha
```
Comment (MrAlias) [inline CHANGELOG.md:297]:
```suggestion
[v0.11.0-alpha]: https://github.com/open-telemetry/opentelemetry-go-instrumentation/releases/tag/v0.11.0-alpha
```

**Comment by damemi** [comment]
@MrAlias should be good now, btw I opened https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/739 to make generating the fixtures a little easier. Links check is going to fail without the new tag.

Should I wait for https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/738 and add that to the changelog?

**Comment by MrAlias** [comment]
> Should I wait for #738 and add that to the changelog?

Yes, please. It should be merged now.

**Comment by MrAlias** [comment]
> Links check is going to fail without the new tag.

Yep, this is the norm :smile:

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ OpenTelemetry Go Automatic Instrumentation adheres to [Semantic Versioning](http
 
 ### Added
 
+### Fixed
```
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
```

Normally we clear the "Unreleased" section.

**Comment by MrAlias** [comment]
> > Should I wait for #738 and add that to the changelog?
> 
> Yes, please. It should be merged now.

Looks like the changelog still needs to be updated.

We should turn on enforcement of the changelog addition that we have in other repos ...

**Comment by damemi** [comment]
@MrAlias updated with a changelog entry for #738 this should be good now

**Review by MrAlias — APPROVED** [review]

**Pull request merged**